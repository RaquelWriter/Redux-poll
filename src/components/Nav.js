import { connect } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { logout } from '../actions/login';

const Nav = (props) => {
  const navigate = useNavigate();
  const { dispatch } = props;
  const { name, avatarURL } = props.userData;

  const handleMenuClick = (e) => {
    e.preventDefault();
    switch (e.target.innerText) {
      case 'Logout':
        dispatch(logout(props.authedUser));
        navigate('/login');
        break;
      case 'Home':
        navigate('/');
        break;
      case 'Leaderboard':
        navigate('/leaderboard');
        break;
      case 'New':
        navigate('/add');
        break;
      default:
        break;
    }
  };
  return (
    <nav>
      <div className='left-nav'>
        <div>
          <span className='nav-link' onClick={handleMenuClick}>
            Home
          </span>
        </div>
        <div>
          <span className='nav-link' onClick={handleMenuClick}>
            Leaderboard
          </span>
        </div>
        <div>
          <span className='nav-link' onClick={handleMenuClick}>
            New
          </span>
        </div>
      </div>
      <div className='right-nav'>
        <div>
          <img className='avatar-small' src={avatarURL} alt='avatar' />
        </div>
        <div>{name}</div>
        <div>
          <button className='nav-button' onClick={handleMenuClick}>
            Logout
          </button>
        </div>
      </div>
    </nav>
  );
};

const mapStateToProps = ({ users, login }) => {
  const authedUser = login.authedUser;
  const userData = users[authedUser];

  return { authedUser, userData };
};

export default connect(mapStateToProps)(Nav);
