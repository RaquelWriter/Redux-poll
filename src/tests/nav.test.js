import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureMockStore from 'redux-mock-store';
import Nav from '../components/Nav';

const mockStore = configureMockStore();

describe('Nav', () => {
  it('will match snapshot if authedUser, name and avatarURL is passed', () => {
    const store = mockStore({
      login: { authedUser: 'saraedo' },
      users: {
        saraedo: {
          name: 'Sara Edo',
          avatarURL: 'https://example.com/avatar.jpg',
        },
      },
    });
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Nav />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
