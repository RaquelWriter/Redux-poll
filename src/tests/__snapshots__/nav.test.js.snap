// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav will match snapshot if authedUser, name and avatarURL is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="left-nav"
        >
          <div>
            <span
              class="nav-link"
            >
              Home
            </span>
          </div>
          <div>
            <span
              class="nav-link"
            >
              Leaderboard
            </span>
          </div>
          <div>
            <span
              class="nav-link"
            >
              New
            </span>
          </div>
        </div>
        <div
          class="right-nav"
        >
          <div>
            <img
              alt="avatar"
              class="avatar-small"
              src="https://example.com/avatar.jpg"
            />
          </div>
          <div>
            Sara Edo
          </div>
          <div>
            <button
              class="nav-button"
            >
              Logout
            </button>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="left-nav"
      >
        <div>
          <span
            class="nav-link"
          >
            Home
          </span>
        </div>
        <div>
          <span
            class="nav-link"
          >
            Leaderboard
          </span>
        </div>
        <div>
          <span
            class="nav-link"
          >
            New
          </span>
        </div>
      </div>
      <div
        class="right-nav"
      >
        <div>
          <img
            alt="avatar"
            class="avatar-small"
            src="https://example.com/avatar.jpg"
          />
        </div>
        <div>
          Sara Edo
        </div>
        <div>
          <button
            class="nav-button"
          >
            Logout
          </button>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
